lib/functions/optimizers.ts

```diff
@@ -37,4 +53,38 @@ export default function gradientDescent(
     return update
 }
 
-export { gradientDescent, mseCostFunction }
+export function updateWeightsAndBias(
+    input: number[][],
+    target: number[],
+    weight: number[],
+    bias: number,
+    learningRate: number,
+    costFunction: Function
+): [number[], number] {
+    const weightInit: number[] = Array(weight.length).fill(0)
+    let weightDeriv = weightInit
+    let biasDeriv = 0
+    let stepSizeWeight = weightInit
+    let stepSizeBias = 0
+    const update: [number[], number] = [weightInit, 0]
```

Comment:
hoist constants at the top of the function.

`const update and const samples` can be below `const weightInit`